instrumentation/logback/logback-appender-1.0/library/src/main/java/io/opentelemetry/instrumentation/logback/appender/v1_0/OpenTelemetryAppender.java

```diff
@@ -69,9 +75,25 @@ public void start() {
     super.start();
   }
 
+  @SuppressWarnings("SystemOut")
   @Override
   protected void append(ILoggingEvent event) {
-    mapper.emit(openTelemetry.getLogsBridge(), event);
+    if (openTelemetry == OpenTelemetry.noop()) {
```

Comment:
I think there's a race condition here that could result in lost logs, e.g.

time=1 `append()` thread gets the no-op instance
time=2 `install()` thread updates the instance and flushes the queue
time=3 `append()` thread adds the log event to the queue

there's some fun related reading in https://github.com/open-telemetry/opentelemetry-java/pull/5976 (very similar problem there, recording metrics in one thread, flushing them in another)

Reply:
Thanks. Would it be possible to address this point in another PR?

Reply:
it's ok for me, it would be good to at least add a TODO here to acknowledge/describe the race condition

Reply:
(to clarify my comment above, I don't think a high-concurrency non-blocking solution like https://github.com/open-telemetry/opentelemetry-java/pull/5976 is needed here, I think some amount of locking is probably ok)

Reply:
> I think some amount of locking is probably ok

If we want to preserve the ordering of the log events then we'll probably have to resort to locking; either that or let all the log events go through a queue

Reply:
I pushed https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/9798/commits/de107e5d992e304b0490c1ab3a2fbd7ac74e8b2b which adds a ReadWriteLock